components/match2/wikis/dota2/matchgroup_input_custom.lua

```diff
@@ -420,17 +419,38 @@ function matchFunctions.isFeatured(match)
 	end
 
 	if
-		(opponent1.type == Opponent.team
-		and Earnings.calculateForTeam({team = opponent1.name, year = year}) >= _MIN_EARNINGS_FOR_FEATURED)
-		or (opponent2.type == Opponent.team
-		and Earnings.calculateForTeam({team = opponent2.name, year = year}) >= _MIN_EARNINGS_FOR_FEATURED)
+		opponent1.type == Opponent.team and
+		matchFunctions.getEarnings(opponent1.name, year) >= _MIN_EARNINGS_FOR_FEATURED
+	then
+		return true
+	elseif
+		opponent2.type == Opponent.team and
+		matchFunctions.getEarnings(opponent2.name, year) >= _MIN_EARNINGS_FOR_FEATURED
 	then
 		return true
 	end
 
 	return false
 end
 
+function matchFunctions.getEarnings(name, year)
+	if String.isEmpty(name) then
+		return 0
+	end
+
+	local data = mw.ext.LiquipediaDB.lpdb('team', {
+		conditions = '[[name::' .. name .. ']]',
+		query = 'extradata'
+	})
+
+	if type(data[1]) == 'table' then
+		local currentEarnings = (data[1].extradata or {})['earningsin' .. year]
+		return tonumber(currentEarnings or 0) or 0
+	end
+
+	return 0
+end
```

Comment:
i suggest to additionally cache it in wikivars so that we can drop the query in case the team is already present in the bracket/matchlist before this match
```suggestion
function matchFunctions.getEarnings(name, year)
	if String.isEmpty(name) then
		return 0
	end

	if String.isNotEmpty(Variables.varDefault(name .. '_featured_earnings')) then
		return tonumber(Variables.varDefault(name .. '_featured_earnings'))
	end

	local data = mw.ext.LiquipediaDB.lpdb('team', {
		conditions = '[[name::' .. name .. ']]',
		query = 'extradata'
	})

	if type(data[1]) == 'table' then
		local currentEarnings = (data[1].extradata or {})['earningsin' .. year]
		,Variables.varDefine(name .. '_featured_earnings', currentEarnings)
		return tonumber(currentEarnings or 0) or 0
	end

	return 0
end
```